apps/meteor/server/modules/watchers/lib/messages.ts

```diff
@@ -19,6 +19,19 @@ export async function getMessageToBroadcast({ id, data }: { id: IMessage['_id'];
 		return;
 	}
 
+	if (message.t) {
+		const hideSystemMessage = (await getSettingCached('Hide_System_Messages')) as MessageTypesValues[];
+
+		if (hideSystemMessage) {
+			const isMutedUnmuted = (messageType: string): boolean => {
+				return messageType === 'user-muted' || messageType === 'user-unmuted';
+			};
+			if (hideSystemMessage.includes(message.t) || (isMutedUnmuted(message.t) && hideSystemMessage.includes('mute_unmute'))) {
```

Comment:
Curious, is this the only case where the setting value is different from the actual message type?

I refer to the message type being `user-muted` but the hide setting having `mute_unmute` 👀